### ISSUE TITLE
find NaN for training faster_rcnn_r101_fpn_1x_coco.py

### ISSUE THREAD

**Issue by clemente0420** [open-mmlab/mmdetection#10183]
find NaN for training faster_rcnn_r101_fpn_1x_coco.py

Package                      Version              Editable project location
---------------------------- -------------------- ------------------------------------------
absl-py                      1.4.0
addict                       2.4.0
aiohttp                      3.8.4
aiosignal                    1.3.1
alembic                      1.10.3
asttokens                    2.2.1
astunparse                   1.6.3
async-timeout                4.0.2
attrs                        22.2.0
backcall                     0.2.0
beautifulsoup4               4.12.1
black                        23.3.0
bleach                       6.0.0
blis                         0.4.1
boto3                        1.14.0
botocore                     1.17.63
cachetools                   5.3.0
certifi                      2022.12.7
chardet                      3.0.4
charset-normalizer           3.1.0
click                        8.1.3
cloudpickle                  2.2.1
colorama                     0.4.6
comm                         0.1.3
contourpy                    1.0.7
coverage                     7.2.2
cpplint                      1.6.1
cycler                       0.11.0
cymem                        2.0.7
databricks-cli               0.17.6
datasets                     2.11.0
debugpy                      1.6.7
decorator                    5.1.1
deepspeed                    0.8.1
defusedxml                   0.7.1
diffusers                    0.14.0
dill                         0.3.6
DLLogger                     1.0.0
docker                       5.0.3
docutils                     0.15.2
EditorConfig                 0.12.3
einops                       0.6.0
en-core-web-sm               2.2.0
entrypoints                  0.4
exceptiongroup               1.1.1
execnet                      1.9.0
executing                    1.2.0
expecttest                   0.1.4
fastjsonschema               2.16.3
filelock                     3.10.7
flake8                       6.0.0
Flask                        2.2.3
flatbuffers                  1.12
fonttools                    4.39.3
frozenlist                   1.3.3
fsspec                       2023.3.0
gast                         0.4.0
gdown                        3.13.0
ghp-import                   2.1.0
gitdb                        4.0.10
GitPython                    3.1.31
google-auth                  2.17.2
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
greenlet                     2.0.2
griffe                       0.26.0
grpcio                       1.53.0
gunicorn                     20.1.0
h5py                         2.10.0
h5sparse                     0.1.0
hjson                        3.1.0
html2text                    2020.1.16
httplib2                     0.22.0
huggingface-hub              0.13.3
hyperparameter               0.4.3
idna                         2.10
imageio                      2.27.0
importlib-metadata           6.1.0
importlib-resources          5.12.0
iniconfig                    2.0.0
ipdb                         0.13.2
ipykernel                    6.22.0
ipython                      8.12.0
ipywidgets                   8.0.6
isort                        5.12.0
itsdangerous                 2.1.2
jedi                         0.18.2
jieba                        0.42.1
Jinja2                       3.1.2
jmespath                     0.10.0
joblib                       1.2.0
jsbeautifier                 1.14.7
jsonschema                   4.17.3
jupyter_client               8.1.0
jupyter_core                 5.3.0
jupyterlab-pygments          0.2.2
jupyterlab-widgets           3.0.7
keras                        2.9.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.4
lark                         1.1.2
lazy_loader                  0.2
libclang                     16.0.0
linformer                    0.2.1
llvmlite                     0.39.1
Mako                         1.2.4
Markdown                     3.3.7
markdown-it-py               2.2.0
MarkupSafe                   2.1.2
matplotlib                   3.7.1
matplotlib-inline            0.1.6
mccabe                       0.7.0
mdurl                        0.1.2
mergedeep                    1.3.4
mirakuru                     2.5.1
mistune                      2.0.5
mkdocs                       1.4.2
mkdocs-autorefs              0.4.1
mkdocs-build-plantuml-plugin 1.7.4
mkdocs-charts-plugin         0.0.8
mkdocs-drawio-exporter       0.8.0
mkdocs-gen-files             0.4.0
mkdocs-glightbox             0.3.2
mkdocs-i18n                  0.4.4
mkdocs-literate-nav          0.6.0
mkdocs-macros-plugin         0.7.0
mkdocs-material              8.4.4
mkdocs-material-extensions   1.1.1
mkdocs-mermaid2-plugin       0.6.0
mkdocstrings                 0.21.1
mkdocstrings-python          0.9.0
mknotebooks                  0.7.1
mlflow                       2.2.2
mmcv-full                    1.7.1
mmengine                     0.7.2
model-index                  0.1.11
multidict                    6.0.4
multiprocess                 0.70.14
murmurhash                   1.0.9
mypy-extensions              1.0.0
nbclient                     0.7.3
nbconvert                    7.3.0
nbformat                     5.8.0
nest-asyncio                 1.5.6
networkx                     3.1
ninja                        1.11.1
nltk                         3.5
numba                        0.56.4
numpy                        1.23.5
oauthlib                     3.2.2
onnx                         1.12.0
onnxruntime                  1.3.0
opencv-python                4.4.0.46
openmim                      0.3.7
opt-einsum                   3.3.0
ordered-set                  4.1.0
packaging                    23.0
pandas                       2.0.0
pandocfilters                1.5.0
parameterized                0.9.0
parso                        0.8.3
pathspec                     0.11.1
pexpect                      4.8.0
pickleshare                  0.7.5
Pillow                       9.5.0
pip                          23.0.1
pkgutil_resolve_name         1.3.10
plac                         1.1.3
platformdirs                 3.2.0
plotly                       5.14.1
pluggy                       1.0.0
port-for                     0.6.3
preshed                      3.0.8
progressbar                  2.5
prompt-toolkit               3.0.38
protobuf                     3.19.6
psutil                       5.9.4
ptyprocess                   0.7.0
pure-eval                    0.2.2
py                           1.11.0
py-cpuinfo                   9.0.0
pyarrow                      11.0.0
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pybind11                     2.9.2
pycocotools                  2.0.5
pycodestyle                  2.10.0
pydantic                     1.10.7
pyflakes                     3.0.1
Pygments                     2.14.0
PyJWT                        2.6.0
pymdown-extensions           9.10
pyparsing                    3.0.9
pyrsistent                   0.19.3
PySocks                      1.7.1
pytest                       7.2.2
pytest-cov                   4.0.0
pytest-cpp                   2.3.0
pytest-html                  3.2.0
pytest-html-merger           0.0.8
pytest-metadata              2.0.4
pytest-picked                0.4.6
pytest-postgresql            4.1.1
pytest-repeat                0.9.1
pytest-timeout               2.1.0
pytest-xdist                 3.2.1
python-dateutil              2.8.2
python-speech-features       0.6
pytz                         2022.7.1
PyWavelets                   1.4.1
PyYAML                       6.0
pyyaml_env_tag               0.1
pyzmq                        25.0.2
querystring-parser           1.2.4
recordclass                  0.18.3
regex                        2023.3.23
requests                     2.23.0
requests-oauthlib            1.3.1
responses                    0.18.0
rich                         13.3.4
rsa                          4.9
s3transfer                   0.3.7
scikit-image                 0.20.0
scikit-learn                 1.2.2
scipy                        1.9.1
seaborn                      0.12.2
semantic-version             2.10.0
sentencepiece                0.1.97
setuptools                   67.6.1
setuptools-rust              1.5.2
shap                         0.41.0
six                          1.15.0
slicer                       0.0.7
smmap                        5.0.0
soupsieve                    2.4
spacy                        2.2.2
SQLAlchemy                   2.0.9
sqlparse                     0.4.3
srsly                        1.0.6
stack-data                   0.6.2
tabulate                     0.9.0
tenacity                     8.2.2
tensorboard                  2.9.1
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorboardX                 2.6
tensorflow                   2.9.0
tensorflow-estimator         2.9.0
tensorflow-io-gcs-filesystem 0.32.0
termcolor                    1.1.0
terminaltables               3.1.10
thinc                        7.3.1
thop                         0.1.1.post2209072238
threadpoolctl                3.1.0
tifffile                     2023.3.21
timm                         0.4.12
tinycss2                     1.2.1
tokenizers                   0.13.3
toml                         0.10.2
tomli                        2.0.1
torch                        1.12.1+cpu
torchtext                    0.4.0
torchvision                  0.13.1+cpu
tornado                      6.2
tqdm                         4.65.0
traitlets                    5.9.0
transformers                 4.27.4
typeguard                    3.0.2
typing_extensions            4.5.0
tzdata                       2023.3
ujson                        5.7.0
Unidecode                    1.3.6
urllib3                      1.25.11
wasabi                       0.10.1
watchdog                     3.0.0
wcwidth                      0.2.6
webencodings                 0.5.1
websocket-client             1.5.1
Werkzeug                     2.2.3
wheel                        0.38.4
widgetsnbextension           4.0.7
wrapt                        1.15.0
xxhash                       3.2.0
yacs                         0.1.8
yapf                         0.32.0
yarl                         1.8.2
zipp                         3.15.0

**Comment by clemente0420** [comment]
# config 

model = dict(
    type='FasterRCNN',
    backbone=dict(
        type='ResNet',
        depth=101,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch',
        init_cfg=dict(type='Pretrained',
                      checkpoint='torchvision://resnet101')),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=5),
    rpn_head=dict(
        type='RPNHead',
        in_channels=256,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            scales=[8],
            ratios=[0.5, 1.0, 2.0],
            strides=[4, 8, 16, 32, 64]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[0.0, 0.0, 0.0, 0.0],
            target_stds=[1.0, 1.0, 1.0, 1.0]),
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
        loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
    roi_head=dict(
        type='StandardRoIHead',
        bbox_roi_extractor=dict(
            type='SingleRoIExtractor',
            roi_layer=dict(type='RoIAlign', output_size=7, sampling_ratio=0),
            out_channels=256,
            featmap_strides=[4, 8, 16, 32]),
        bbox_head=dict(
            type='Shared2FCBBoxHead',
            in_channels=256,
            fc_out_channels=1024,
            roi_feat_size=7,
            num_classes=80,
            bbox_coder=dict(
                type='DeltaXYWHBBoxCoder',
                target_means=[0.0, 0.0, 0.0, 0.0],
                target_stds=[0.1, 0.1, 0.2, 0.2]),
            reg_class_agnostic=False,
            loss_cls=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
            loss_bbox=dict(type='L1Loss', loss_weight=1.0))),
    train_cfg=dict(
        rpn=dict(
            assigner=dict(
                type='MaxIoUAssigner',
                pos_iou_thr=0.7,
                neg_iou_thr=0.3,
                min_pos_iou=0.3,
                match_low_quality=True,
                ignore_iof_thr=-1),
            sampler=dict(
                type='RandomSampler',
                num=256,
                pos_fraction=0.5,
                neg_pos_ub=-1,
                add_gt_as_proposals=False),
            allowed_border=-1,
            pos_weight=-1,
            debug=False),
        rpn_proposal=dict(
            nms_pre=2000,
            max_per_img=1000,
            nms=dict(type='nms', iou_threshold=0.7),
            min_bbox_size=0),
        rcnn=dict(
            assigner=dict(
                type='MaxIoUAssigner',
                pos_iou_thr=0.5,
                neg_iou_thr=0.5,
                min_pos_iou=0.5,
                match_low_quality=False,
                ignore_iof_thr=-1),
            sampler=dict(
                type='RandomSampler',
                num=512,
                pos_fraction=0.25,
                neg_pos_ub=-1,
                add_gt_as_proposals=True),
            pos_weight=-1,
            debug=False)),
    test_cfg=dict(
        rpn=dict(
            nms_pre=1000,
            max_per_img=1000,
            nms=dict(type='nms', iou_threshold=0.7),
            min_bbox_size=0),
        rcnn=dict(
            score_thr=0.05,
            nms=dict(type='nms', iou_threshold=0.5),
            max_per_img=100)))
dataset_type = 'CocoDataset'
data_root = 'data/coco/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='CocoDataset',
        ann_file='data/coco/annotations/instances_train2017.json',
        img_prefix='data/coco/train2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
        ]),
    val=dict(
        type='CocoDataset',
        ann_file='data/coco/annotations/instances_val2017.json',
        img_prefix='data/coco/val2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='CocoDataset',
        ann_file='data/coco/annotations/instances_val2017.json',
        img_prefix='data/coco/val2017/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
evaluation = dict(interval=1, metric='bbox')
optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.001,
    step=[8, 11])
runner = dict(type='EpochBasedRunner', max_epochs=12)
checkpoint_config = dict(interval=1)
log_config = dict(interval=50, hooks=[dict(type='TextLoggerHook')])
custom_hooks = [dict(type='NumClassCheckHook')]
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
opencv_num_threads = 0
mp_start_method = 'fork'
auto_scale_lr = dict(enable=True, base_batch_size=16)
work_dir = './work_dirs/faster_rcnn_r101_fpn_1x_coco'
auto_resume = False
gpu_ids = [0]
